### PR TITLE
Feature: New proposal flow

### PR DESCRIPTION
**Pull request by aidan-starke**
## Why

 - Set up form for submitting a new proposal

## What is changing

 - Re-export vectors & components from their folders via `index.ts`
 - Extract `WalletConnect` to own file
 - Add `react-markdown` & `remark-gfm` dependencies to parse & display markdown

### CODE REVIEW

File: packages/gov-app/libs/components/ProposalAdvanced.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+			</div>
+			<br />
+			<If condition={open}>
+				<label htmlFor="cennzExtrinsic" className="text-lg">
+					Extrinsic Hash
+				</label>
+				<TextField
+					id="cennzExtrinsic"
+					inputClassName="w-full truncate"
+					value={extrinsicHash}
```
Comment (ken-futureverse) [inline packages/gov-app/libs/components/ProposalAdvanced.tsx]: this should be `multiline`

File: packages/gov-app/libs/components/WalletConnect.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
+import type { ChangeEventHandler, FC, MouseEventHandler } from "react";
```
Comment (ken-futureverse) [inline packages/gov-app/libs/components/WalletConnect.tsx:1]: good idea, but we should treat this as an enhanced form field, say `WalletField` which should contain only the `Select` and all of the logic to have it working as a standard form field.

The rest of the stuff, e.g the `h2` and `p` contents should belong to the page, as we cannot guarantee all pages will have to use the exact same content.

File: packages/gov-app/libs/components/index.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export { Button } from "@gov-app/libs/components/Button";
+export { Header } from "@gov-app/libs/components/Header";
+export { Layout } from "@gov-app/libs/components/Layout";
+export { ProposalAdvanced } from "@gov-app/libs/components/ProposalAdvanced";
+export { ProposalDetails } from "@gov-app/libs/components/ProposalDetails";
+export { Select } from "@gov-app/libs/components/Select";
+export { TextField } from "@gov-app/libs/components/TextField";
+export { WalletConnect } from "@gov-app/libs/components/WalletConnect";
```
Comment (ken-futureverse) [inline packages/gov-app/libs/components/index.ts:8]: thanks for doing this, will have a hygen template to handle this to ease the pain :)

File: packages/gov-app/pages/proposals/new.tsx [hunk 4]
```diff
@@ -1,16 +1,108 @@
...
+	ProposalAdvanced,
+	ProposalDetails,
+	TextField,
+	WalletConnect,
+} from "@gov-app/libs/components";
+import { Spinner } from "@gov-app/libs/assets/vectors";
+import { ChangeEvent, FormEventHandler, useCallback, useState } from "react";
 
 const NewProposal: NextPage = () => {
+	const [proposalTitle, setProposalTitle] = useState<string>("");
```
Comment (ken-futureverse) [inline packages/gov-app/pages/proposals/new.tsx]: why do we need a state for `proposalTitle`?

File: packages/gov-app/pages/proposals/new.tsx [hunk 5]
```diff
@@ -1,16 +1,108 @@
...
-			<div className="flex flex-1 items-center justify-center">
-				<h1 className="font-display text-6xl uppercase">New Proposal</h1>
+			<div className="w-full max-w-3xl flex-1 self-center px-8 pb-12">
+				<form onSubmit={onFormSubmit}>
+					<h1 className="font-display mb-8 text-center text-7xl uppercase">
+						Submit a Proposal
+					</h1>
+
+					<p className="mb-8 text-center text-lg">
+						To submit a proposal you must be a CENNZnet Councillor.
```
Comment (ken-futureverse) [inline packages/gov-app/pages/proposals/new.tsx]: need to extend this a bit, can you append the paragraph with "lorem ipsum" text to make it as place holder for now?

File: packages/gov-app/libs/components/ProposalDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+import { Button, TextField } from "@gov-app/libs/components";
+
+export const ProposalDetails: FC = () => {
+	const [showPreview, setShowPreview] = useState<boolean>();
+	const [proposalDetails, setProposalDetails] = useState<string>();
+
+	return (
+		<div className="w-full">
+			<div className="float-right mr-[1px] inline-flex" role="group">
+				<Button size="small" onClick={() => setShowPreview(false)}>
```
Comment (ken-futureverse) [inline packages/gov-app/libs/components/ProposalDetails.tsx]: needs an `active` state here to tell which mode you are in

File: packages/gov-app/libs/components/ProposalDetails.tsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+						setProposalDetails(event.target.value)
+					}
+					multiline
+					required
+				/>
+			</If>
+			<If condition={showPreview}>
+				<div className="border-dark flex w-full border-[3px] bg-white px-4 py-2">
+					<ReactMarkdown remarkPlugins={[[RemarkGfm, { singleTilde: false }]]}>
+						{proposalDetails}
```
Comment (ken-futureverse) [inline packages/gov-app/libs/components/ProposalDetails.tsx]: should show "Nothing to preview" rather an empty white box

**Review by ken-futureverse — CHANGES_REQUESTED** [review]
looks good @aidan-starke, and assuming you want to do the form submission as a separate PR?

File: packages/gov-app/pages/proposals/new.tsx (hunk 4 above)
Comment (aidan-starke) [reply packages/gov-app/pages/proposals/new.tsx]: For the justification metadata

File: packages/gov-app/libs/components/ProposalAdvanced.tsx (hunk 1 above)
Comment (aidan-starke) [reply packages/gov-app/libs/components/ProposalAdvanced.tsx]: I have `truncate` which appends an ellipsis for any overflow, but can change if you prefer

**Pull request merged**